Formula/boringtun.rb

```diff
@@ -0,0 +1,17 @@
+class Boringtun < Formula
+  desc "Userspace WireGuard implementation in Rust"
+  homepage "https://github.com/cloudflare/boringtun"
+  url "https://github.com/cloudflare/boringtun/archive/v0.2.0.tar.gz"
+  sha256 "544c72fc482b636e7f6460bfee205adafc55de534067819e4e4914980f0d1350"
+  head "https://github.com/cloudflare/boringtun.git"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
+  end
+
+  test do
+    system "#{bin}/boringtun", "--help"
```

Comment:
This is not a sufficient test. We need tests that are more than just `binary --help` and `binary --version`...

Can you please add a better test here, and squash the new commit into the old one?

Thanks!

Reply:
@zbeekman As I said:
> I'm not aware of any functionality that does not involve root privileges and the creation of a network interface, so currently the test is simply running boringtun --help. I'm happy to update it if there's something better that can be done.

Reply:
Sorry I missed that comment. Despite this, there are things we can do to further test functionality, even if we cannot open a new utun connection and we don't have root:

```suggestion
    system "#{bin}/boringtun", "--help"
    assert_match "boringtun " + version.to_s, shell_output("#{bin}/boringtun -V").chomp
    assert_match /failed to start/, shell_output("#{bin}/boringtun utun --log #{testpath}/boringtun.log")
    assert_predicate testpath/"boringtun.log" :exist?
    boringtun_log = File.read(testpath/"boringtun.log")
    assert_match /Success, daemonized/, boringtun_log[1]
```